src/qtils/hex.hpp

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright Quadrivium LLC
+ * All Rights Reserved
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include <fmt/format.h>
+
+#include <qtils/bytes.hpp>
+
+template <>
+struct fmt::formatter<qtils::BytesIn> {
+  bool prefix = true;
+  bool full = false;
+  constexpr auto parse(format_parse_context &ctx) {
+    auto it = ctx.begin();
+    auto end = [&] { return it == ctx.end() or *it == '}'; };
+    if (end()) {
+      return it;
+    }
+    if (*it == 'x') {
+      ++it;
+      prefix = false;
+      full = true;
+      if (end()) {
+        return it;
+      }
+    } else if (*it == '0') {
+      ++it;
+      prefix = true;
+      if (not end() and *it == 'x') {
+        ++it;
+        full = true;
+        if (end()) {
+          return it;
+        }
+      }
+    }
+    fmt::throw_format_error("\"x\" or \"0x\" expected");
+  }
+  auto format(const qtils::BytesIn bytes, format_context &ctx) const {
+    auto out = ctx.out();
+    if (prefix) {
+      out = fmt::format_to(out, "0x");
```

Comment:
Can we write "0x" directly?

Reply:
Thanks, used `fmt::detail::write(output_iterator, str)`